### PR TITLE
Vehicle testing

### PR DESCRIPTION
**Pull request by RTO-Chatter**
-  Added Load and Save for Vehicles.
-  Vehicle interaction options have their own separate settings
-  Settings are applied to the currently active radio

### CODE REVIEW

**Comment by RTO-Chatter** [comment]
Pull Request Issue:
- Testing code still inside, have to re-implement conditions for tree settings.
# Do not merge at this time

**Comment by RTO-Chatter** [comment]
Good for review now.

**Comment by RTO-Chatter** [comment]
I'm Bad and Messed up the loading of SR and the tree for vehicle loading and didn't catch it. Fixed in e150b62 push. Did a full test of everything just to make sure I gots it all.

File: CHTR_TFAR_Setter/functions/fn_getRadioData.sqf [hunk 1]
```diff
@@ -3,7 +3,11 @@
  * Gets either LR or SR radio data from the settings array and returns it
  *
  * Arguments:
- * 0: LR if true, SR if False <BOOLEAN>
+ * 0: LR, SR, or vehicleLR (2, 3, 5) (default:2) <INTEGER>
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: Bad merge, git headers, cleanup

File: CHTR_TFAR_Setter/CfgVehicles.hpp [hunk 2]
```diff
@@ -69,42 +69,77 @@ class CfgVehicles {
...
 					};
+          
 					//New Layout, ALTERNATE_LAYOUT = True
 					class Load_Root {
 						displayName = "Load";
 						icon = QUOTE(ICON_PATH(load));
 						statement = QUOTE(if(SHORTCUT_ENABLED && SHOW_LR && SHOW_SR) then {[true] call FUNC(loadBothSettings)};);
-						condition = QUOTE(ALTERNATE_LAYOUT && (HAS_LR || HAS_SR));
+<<<<<<< HEAD
+						condition = QUOTE(ALTERNATE_LAYOUT);
```
Comment (M3ales) [inline CHTR_TFAR_Setter/CfgVehicles.hpp]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_getRadioData.sqf [hunk 3]
```diff
@@ -15,7 +19,11 @@
  */
 #include "function_macros.hpp"
 params[
-	["_lr", true, [true]]
+	["_radio", 2, [1.0]]
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_getRadioData.sqf (hunk 1 above)
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: Why 2 5 7 and not 0 1 2? For LR param?

File: CHTR_TFAR_Setter/functions/fn_getRadioData.sqf [hunk 4]
```diff
@@ -26,9 +34,12 @@ if(count _settings == 0) exitWith {
...
 
-_index = 2;
-if(!_lr) then {
-	_index = 3;
-};
-
-_currentProfile select _index
\ No newline at end of file
+/*if(_radio = 3) then {
+	LOG("Fetching SR Radio Data");
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: Legacy code, big delete, have git history

File: CHTR_TFAR_Setter/functions/fn_loadLRSettings.sqf [hunk 5]
```diff
@@ -21,7 +21,11 @@
 params[["_showResult", true, [true]]];
 
 LOG("Loading LR Settings");
-_radioData = [true] call FUNC(getRadioData);
+_radioData = [2] call FUNC(getRadioData);
+<<<<<<< HEAD
+=======
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_loadLRSettings.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_loadSRSettings.sqf [hunk 6]
```diff
@@ -21,7 +21,11 @@
 params[["_showResult", true, [true]]];
 
 LOG("Loading SR Settings");
-_radioData = [false] call FUNC(getRadioData);
+_radioData = [3] call FUNC(getRadioData);
+<<<<<<< HEAD
+=======
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_loadSRSettings.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_saveLRSettings.sqf [hunk 7]
```diff
@@ -22,7 +22,11 @@ params[["_showResult", true, [true]]];
 
 LOG("Saving LR Settings");
 _radioData = (call TFAR_fnc_activeLrRadio) call TFAR_fnc_getLrSettings;
-[true, _radioData] call FUNC(setRadioData);
+[2, _radioData] call FUNC(setRadioData);
+<<<<<<< HEAD
+=======
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_saveLRSettings.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_saveSRSettings.sqf [hunk 8]
```diff
@@ -22,7 +22,11 @@ params[["_showResult", true, [true]]];
 
 LOG("Saving SR Settings");
 _radioData = (call TFAR_fnc_activeSwRadio) call TFAR_fnc_getSwSettings;
-[false, _radioData] call FUNC(setRadioData);
+[3, _radioData] call FUNC(setRadioData);
+<<<<<<< HEAD
+=======
+
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_saveSRSettings.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_saveVLRSettings.sqf [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ * 
+ *
+ * Arguments:
+ * _showResult (optional): If true will display a message at the top right using ace_common_fnc_displayTextPicture on success
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [true] call CHTR_TFAR_Setter_fnc_saveLRSettings
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_saveVLRSettings.sqf]: VLRSettings

File: CHTR_TFAR_Setter/functions/fn_setRadioData.sqf [hunk 10]
```diff
@@ -3,7 +3,11 @@
  * Sets LR/SR data of current profile
  *
  * Arguments:
- * 0: LR or SR, LR is true (default:true) <BOOLEAN>
+ * 0: LR, SR, or vehicleLR (2, 3, 5) (default:2) <INTEGER>
+<<<<<<< HEAD
+=======
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_setRadioData.sqf]: Bad merge git headers, cleanup

File: CHTR_TFAR_Setter/functions/fn_loadVLRSettings.sqf [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+ * _showResult (optional): If true will display a message at the top right using ace_common_fnc_displayTextPicture on success
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [true] call CHTR_TFAR_Setter_fnc_loadLRSettings
+ *
+ * Public: No
+ */
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_loadVLRSettings.sqf]: Take a look at [tfar_fnc_vehicleLR](https://github.com/michail-nikolaev/task-force-arma-3-radio/blob/f6fb204f9cdd6abe6f95b039f81ac3fbb526feb6/arma3/%40task_force_radio/addons/task_force_radio/functions/fn_vehicleLr.sqf#L18) maybe cleaner implementation.

File: CHTR_TFAR_Setter/functions/fn_saveVLRSettings.sqf [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Public: No
+ */
+#include "function_macros.hpp"
+
+params[["_showResult", true, [true]]];
+
+LOG("Saving Vehicle LR Settings");
+_radioData = (call TFAR_fnc_activeLrRadio) call TFAR_fnc_getLrSettings;
```
Comment (M3ales) [inline CHTR_TFAR_Setter/functions/fn_saveVLRSettings.sqf]: Read comment about vehicleLR thing in the load version.

**Comment by M3ales** [comment]
[Get Vehicle LR Specifically](https://github.com/michail-nikolaev/task-force-arma-3-radio/blob/0.9.7/arma3/%40task_force_radio/addons/task_force_radio/functions/fn_vehicleLr.sqf). 

Better implementation, can do checking for if [has vehicle LR](https://github.com/michail-nikolaev/task-force-arma-3-radio/blob/0.9.7/arma3/%40task_force_radio/addons/task_force_radio/functions/fn_hasVehicleRadio.sqf)

File: CHTR_TFAR_Setter/functions/fn_getRadioData.sqf (hunk 1 above)
Comment (RTO-Chatter) [reply CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: 2 and 3 are already LR and SR, 4 is used by... CBA settings I believe, so VLR would be 5

Comment (M3ales) [reply CHTR_TFAR_Setter/functions/fn_getRadioData.sqf]: Make param accept 012 and modify self, method is less confusing then.

**Comment by M3ales** [comment]
Closed for rework

**Pull request closed without merging**